storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6327,17 +6326,22 @@ int ha_rocksdb::read_row_from_secondary_key(uchar *const buf,
   int rc = 0;
   uint pk_size;
 
-  if (m_keyread_only && m_lock_rows == RDB_LOCK_NONE && !has_hidden_pk(table)) {
-    /* Get the key columns and primary key value */
-    const rocksdb::Slice &rkey = m_scan_it->key();
+  /* Get the key columns and primary key value */
+  const rocksdb::Slice &rkey = m_scan_it->key();
+  const rocksdb::Slice &value = m_scan_it->value();
+
+  bool covered_lookup = m_keyread_only || kd.covers_lookup(table, &value);
+  if (covered_lookup && m_lock_rows == RDB_LOCK_NONE && !has_hidden_pk(table)) {
     pk_size =
         kd.get_primary_key_tuple(table, *m_pk_descr, &rkey, m_pk_packed_tuple);
-    const rocksdb::Slice &value = m_scan_it->value();
     if (pk_size == RDB_INVALID_KEY_LEN) {
       rc = HA_ERR_ROCKSDB_CORRUPT_DATA;
     } else {
       rc = kd.unpack_record(table, buf, &rkey, &value,
                             m_verify_row_debug_checksums);
+      if (!rc) {
+        global_stats.covered_secondary_key_lookups.inc();
```

Comment:
Ah, okay, that sounds good then. We could also bump the count regardless of rc to indicate covered_sk lookup attempts. rc would be non-zero in the case of key/value corruption or I/O error, which should be rare. I realize the row read counters are only incremented on rc == 0. Either way works for me.